truffleruby/src/main/java/org/truffleruby/language/arguments/ReadCallerFrameNode.java

```diff
@@ -51,9 +51,9 @@ private void replaceDispatchNode() {
             Node parent = callerNode.getParent();
             if (parent instanceof CachedDispatchNode) {
                 if (getContext().getCallStack().callerIsSend()) {
-                    ((CachedDispatchNode) parent).replaceSendingCallerFrame(accessMode);
+                    ((CachedDispatchNode) parent).startSendingCallerFrame(accessMode);
                 } else {
-                    ((CachedDispatchNode) parent).replaceSendingFrame();
+                    ((CachedDispatchNode) parent).startSendingOwnFrame();
                 }
```

Comment:
Would it be easier to let the CachedDispatchNode handle this by just checking the cached InternalMethod?